actioncable/app/javascript/action_cable/subscriptions.js

```diff
@@ -29,10 +29,15 @@ export default class Subscriptions {
   // Private
 
   add(subscription) {
+    const existing = this.findAll(subscription.identifier).length > 0
     this.subscriptions.push(subscription)
     this.consumer.ensureActiveConnection()
     this.notify(subscription, "initialized")
-    this.subscribe(subscription)
+    if (!existing) {
+      this.subscribe(subscription)
+    } else if (!this.guarantor.pendingSubscriptions.some((s) => s.identifier === subscription.identifier)) {
```

Comment:
I'd add a method to `SubscriptionGuarantor` to avoid having to access `pendingSubscriptions` from the outside. E.g: `!this.guarantor.hasPendingSubcription(subscription.identifier)`